src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -199,8 +197,7 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                     writer.Line($"{objAdditionalProperties.Type} {dictionaryVariable:D} = new {objAdditionalProperties.Type}();");
                 }
 
-                writer.Line($"foreach (var {itemVariable:D} in {element}.EnumerateObject())");
-                using (writer.Scope())
+                using (writer.Scope($"foreach (var {itemVariable:D} in {element}.EnumerateObject())"))
```

Comment:
This change makes so `itemVariable` is defined inside foreach scope and results in better names.